### PR TITLE
Allow to exclude host from being proxying #2600

### PR DESCRIPTION
**Pull request by KowalczykBartek**
Initial PR for #2600

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/HttpClientImpl.java [hunk 1]
```diff
@@ -1030,6 +1030,17 @@ private HttpClientRequest createRequest(HttpMethod method, String protocol, Stri
     return req;
   }
 
+  private boolean useProxy(boolean isSsl, ProxyOptions proxyOptions, String targetHost) {
```
Comment (xkr47) [inline src/main/java/io/vertx/core/http/impl/HttpClientImpl.java]: This functionality is duplicated in `HttpChannelConnector.doConnect()` that you also updated. Please move this helper method to e.g. a new class `src/main/java/io/vertx/core/http/impl/ProxyHelper.java`. Then reuse that helper in `doConnect()` as well. (You'll have to extract the proxy type frokm the proxyOptions instead, but that's fine.)

File: src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java [hunk 2]
```diff
@@ -129,7 +137,7 @@ private void doConnect(
 
     ChannelProvider channelProvider;
     // http proxy requests are handled in HttpClientImpl, everything else can use netty proxy handler
-    if (options.getProxyOptions() == null || !ssl && options.getProxyOptions().getType()== ProxyType.HTTP ) {
+    if (options.getProxyOptions() == null || !ssl && options.getProxyOptions().getType() == ProxyType.HTTP || isExcluded(options.getProxyOptions(), host)) {
```
Comment (xkr47) [inline src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java]: See comment for `HttpClientImpl.useProxy()`.

File: src/main/java/io/vertx/core/net/ProxyOptions.java [hunk 3]
```diff
@@ -46,6 +46,7 @@
   private String username;
   private String password;
   private ProxyType type;
+  private Set<String> exludedHosts;
```
Comment (xkr47) [inline src/main/java/io/vertx/core/net/ProxyOptions.java]: Typo - please rename to `excludedHosts`.

File: src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java [hunk 4]
```diff
@@ -116,6 +117,13 @@ public void deactivate(HttpClientConnection conn) {
     }
   }
 
+  private boolean isExcluded(ProxyOptions options, String targetProxy) {
```
Comment (xkr47) [inline src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java]: Remove this as it's no longer needed; see comment for `HttpClientImpl.useProxy()`.

**Comment by KowalczykBartek** [comment]
Thanks @xkr47 for the review - unfortunately, this PR will never be merged - I will close it to note confuse anyone .

**Pull request closed without merging**